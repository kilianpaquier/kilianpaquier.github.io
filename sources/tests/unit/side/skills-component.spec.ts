import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import SkillComponent from "@/components/side/SkillComponent.vue";
import Vue from "vue";
import Vuetify from "vuetify";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("SkillComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let wrapper: Wrapper<any>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(SkillComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create SkillComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("should return 12 with a list of 1 elements with index 0", () => {
        expect(wrapper.vm.lastCol(0, ["test"])).toEqual(12);
    });

    it("should return 6 with a list of 2 elements with index 0", () => {
        expect(wrapper.vm.lastCol(0, ["test", "test"])).toEqual(6);
    });

    it("should return 6 with a list of 2 elements with index 1", () => {
        expect(wrapper.vm.lastCol(1, ["test", "test"])).toEqual(6);
    });

    it("should return 12 with a list of 3 elements with index 2", () => {
        expect(wrapper.vm.lastCol(2, ["test", "test", "test"])).toEqual(12);
    });
});
