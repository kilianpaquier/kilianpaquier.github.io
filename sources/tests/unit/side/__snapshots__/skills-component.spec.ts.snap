// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillComponent.vue should create SkillComponent 1`] = `
<div class="container container--fluid">
  <h2 class="font-weight-bold">$vuetify.skills-title</h2>
  <div role="list" class="v-list pt-0 v-sheet theme--light">
    <div tabindex="-1" role="listitem" class="text-right pe-0 v-list-item theme--light mb-n3">
      <div class="v-list-item__content">
        <div class="v-list-item__title">$vuetify.soft-skills.one</div>
      </div>
    </div>
    <div tabindex="-1" role="listitem" class="text-right pe-0 v-list-item theme--light">
      <div class="v-list-item__content">
        <div class="v-list-item__title">$vuetify.soft-skills.two</div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>Java</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="70" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 70%; width: 30%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 70%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-spring-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>Frontend</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="70" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 70%; width: 30%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 70%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-typescript-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-nestjs-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-angularjs-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-sass-original"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-jquery-plain"></em>
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>$vuetify.hard-skills.tests</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="70" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 70%; width: 30%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 70%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-cucumber-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-jasmine-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-jest-plain"></em>
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>Cloud</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="50" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 50%; width: 50%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 50%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-kubernetes-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-prometheus-original"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>$vuetify.hard-skills.database</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="60" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 60%; width: 40%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 60%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>CI/CD</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="60" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 60%; width: 40%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 60%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>$vuetify.hard-skills.elastic-stack</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="30" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 30%; width: 70%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 30%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>Python</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="40" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 40%; width: 60%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 40%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-flask-plain"></em>
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>C#</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="40" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 40%; width: 60%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 40%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-6 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>C/C++</span>
          <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="30" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 6px;">
            <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 30%; width: 70%;"></div>
            <div class="v-progress-linear__buffer"></div>
            <transition-stub name="slide-x-transition">
              <div class="v-progress-linear__determinate primary" style="width: 30%;"></div>
            </transition-stub>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-qt-original"></em>
          <!---->
        </div>
      </div>
    </div>
    <div class="col-sm-12 col-md-12 col-lg-12 col-xl-12 col-12">
      <div class="row no-gutters">
        <div class="mb-1 col col-12"><span>$vuetify.hard-skills.tools</span>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-bash-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-yarn-plain"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span><em aria-haspopup="true" aria-expanded="false" class="small-icon colored devicon-npm-original-wordmark"></em>
          <!---->
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="pa-1 col col-auto">
          <!----><span class="v-tooltip v-tooltip--bottom"><!----></span>
          <!---->
          <div class="v-image v-responsive mx-auto small-img theme--light" aria-haspopup="true" aria-expanded="false">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
